### PR TITLE
 #34302 add ipv6 square bracket support to launch_on_machine 

### PR DESCRIPTION
**Pull request by ssikdar1**
PR from #34430 for issue #34302 

With an accidental reset and force push, I closed  PR 34430 and seem unable to reopen it.
So I'm creating another PR here.

### CODE REVIEW

File: stdlib/Distributed/src/managers.jl [hunk 1]
```diff
@@ -145,7 +145,42 @@ end
 show(io::IO, manager::SSHManager) = println(io, "SSHManager(machines=", manager.machines, ")")
 
 
-function launch_on_machine(manager::SSHManager, machine, cnt, params, launched, launch_ntfy::Condition)
+function parse_machine(machine::AbstractString)
+    hoststr = ""
+    portnum = nothing
+
+    if machine[begin] == '['  # ipv6 bracket notation (RFC 2732)
+        ipv6_end = findlast(']', machine)
```
Comment (fredrikekre) [inline stdlib/Distributed/src/managers.jl]: Should this check for `ipv6_end === nothing`?

File: stdlib/Distributed/src/managers.jl [hunk 2]
```diff
@@ -145,7 +145,42 @@ end
...
+        hoststr = machine[begin+1 : ipv6_end - 1]
```
Comment (fredrikekre) [inline stdlib/Distributed/src/managers.jl]: Should this use `prevind`/`lastind`?

File: stdlib/Distributed/src/managers.jl [hunk 3]
```diff
@@ -145,7 +145,42 @@ end
...
+
+    if length(machine_def) == 2
+        portstr = machine_def[2]
+
+        if !all(isdigit, portstr)
+            msg = "invalid machine definition format string: invalid port format \"$machine_def\""
+            throw(ArgumentError(msg))
+        end
+
+        portnum = parse(Int, portstr)
```
Comment (fredrikekre) [inline stdlib/Distributed/src/managers.jl]: Maybe use `tryparse` and check for `nothing`?

File: stdlib/Distributed/test/managers.jl [hunk 4]
```diff
@@ -0,0 +1,19 @@
+using Test
+using Distributed
+using Base: Semaphore
+using Sockets
```
Comment (fredrikekre) [inline stdlib/Distributed/test/managers.jl]: These don't seem to be used?

File: stdlib/Distributed/test/runtests.jl [hunk 5]
```diff
@@ -3,9 +3,11 @@
 # Run the distributed test outside of the main driver since it needs its own
 # set of dedicated workers.
 include(joinpath(Sys.BINDIR, "..", "share", "julia", "test", "testenv.jl"))
+
 disttestfile = joinpath(@__DIR__, "distributed_exec.jl")
+managerstestfile = joinpath(@__DIR__, "managers.jl")
 
-cmd = `$test_exename $test_exeflags $disttestfile`
+cmd = `$test_exename $test_exeflags $disttestfile $managerstestfile`
```
Comment (fredrikekre) [inline stdlib/Distributed/test/runtests.jl]: I think we can just include this file on its own (e.g. `include("managers.jl")`) on line 15. Does not have to run in a separate process etc.

**Comment by ssikdar1** [comment]
ok! Also @StefanKarpinski tagging you here, since you were the main review of the last PR as well.

File: stdlib/Distributed/src/managers.jl (hunk 1 above)
Comment (ssikdar1) [reply stdlib/Distributed/src/managers.jl]: addressed in 894c51e

File: stdlib/Distributed/src/managers.jl (hunk 3 above)
Comment (ssikdar1) [reply stdlib/Distributed/src/managers.jl]: addressed in 31aef63

File: stdlib/Distributed/test/managers.jl (hunk 4 above)
Comment (ssikdar1) [reply stdlib/Distributed/test/managers.jl]: removed in 4f1f3c9

File: stdlib/Distributed/test/runtests.jl (hunk 5 above)
Comment (ssikdar1) [reply stdlib/Distributed/test/runtests.jl]: addressed in 4f1f3c9

File: stdlib/Distributed/src/managers.jl (hunk 2 above)
Comment (ssikdar1) [reply stdlib/Distributed/src/managers.jl]: addressed in 7371125

**Comment by ssikdar1** [comment]
@fredrikekre @StefanKarpinski 
Can't tell if the win32 win64 build failure are related to my changes or not? But they appear to be not related

**Comment by StefanKarpinski** [comment]
Win32 is a timeout so I restarted it; Win64 is a known, unrelated issue.

**Review by StefanKarpinski — APPROVED** [review]
This seems good to go.

**Pull request merged**